### ISSUE TITLE
How to announce/discuss new reports

### ISSUE THREAD

**Issue by jonathanmetzman** [google/fuzzbench#205]
Right now when I run a new experiment, I typically reply to old issues to discuss results in the experiment. I think this isn't ideal.
What do people think of having a set place to discuss results? Some ideas for this I have are:
1. Automatically announcing new experiments on [fuzzbench-users mailing list](https://groups.google.com/g/fuzzbench-users). This might make the list too noisy though.
2. Manually announcing new experiments in fuzzbench-users (less noise in mailinglist, more work for me).
3. Creating an issue to discuss each new experiment.

Users of fuzzbench (e.g. @vanhauser-thc @andreafioraldi @robertswiecki @thuanpv @alifahmed @inferno-chromium @lszekeres) please share your thoughts on this.

**Comment by robertswiecki** [comment]
Just a random thought - maybe a forever-opened bug where people can subscribe for updates? Can be either manual or automatic updates. Though sending it each time to fuzzbench-users can also work, I can always filter those into some maildir folder if they became too noisy.

**Comment by vanhauser-thc** [comment]
commenting on issues and PRs can be fine if it is concerning one user/target/fuzzer.
However things that affect everyone, e.g. offering a different, newer image, llvm 11, additional measurement for fuzzing, etc. - that is better on a mailing list (IMHO)

**Comment by lszekeres** [comment]
Another option is to do the following two things at the same time:
- Automatically announce each new experiment result in an always open github issue, as @robertswiecki suggested.
- Every month or so we write and post a quick summary of the latest results and major changes to the mailing list.

**Comment by andreafioraldi** [comment]
You can also (ab)use releases. GitHub offers "watch only releases" and you can release fuzzbench each time there is a new report adding the link to a md file. You can ofc easily write a script to do that.

**Comment by mboehme** [comment]
Some info on the changes versus the most recent report would be good (e.g., added/updated fuzzers and configurations; which changes worked, which didn't work). To source this information, we could ask the commit-author to provide information on the recent updates to the fuzzer (and the expected results)? You already fix a fuzzer's commit hash in the builder.Dockerfile, so upstream changes to the fuzzer should be visible to fuzzbench by updates to that commit hash.

**Comment by andreafioraldi** [comment]
In https://www.fuzzbench.com/reports/2020-04-08-aflplusplus-and-2020-04-01/index.html the best schedule seems to be RARE, but if we rate the fuzzer not by the place reached in each target (as currently fuzzbench does) but giving to fuzzers, for each target, a score that is the percentage of the edge coverage of the best fuzzer for the target Marc fond that the best schedule, the one that is "good almost for all" is COE.

**Comment by vanhauser-thc** [comment]
yeah that was a surprise to me - and a disappointment. but at least RARE and MMOPT, my own new creations, were place 2 and 3.
Still have to see where the default EXPLORE schedule will fit in. my guess it is place 4 or 5.

I hope fuzzbench will adopt that ranking (% of edges found compared to the best one), as it is way better.  e.g. when looking at the outcome of jsoncpp_jsoncpp_fuzzer, the best fuzzer found 635 edges, and 80% of the other fuzzers found 634.5 to 633.5. edges. that is so close together that a rank 1 or rank 6 is pretty meaningless. 

Just give every fuzzer 0-100 points depending on their percentage of edges found vs. the best and sum them all up = a real picture which fuzzer is the best overall. I think honggfuzz and perhaps entropic would benefit most from that calculation.

**Comment by jonathanmetzman** [comment]
> Another option is to do the following two things at the same time:
> 
> * Automatically announce each new experiment result in an always open github issue, as @robertswiecki suggested.
> * Every month or so we write and post a quick summary of the latest results and major changes to the mailing list.

Thanks everyone for the ideas. I still haven't decided but I'm leaning towards something like this for the time being.

**Comment by jonathanmetzman** [comment]
> Some info on the changes versus the most recent report would be good (e.g., added/updated fuzzers and configurations; which changes worked, which didn't work). To source this information, we could ask the commit-author to provide information on the recent updates to the fuzzer (and the expected results)? You already fix a fuzzer's commit hash in the builder.Dockerfile, so upstream changes to the fuzzer should be visible to fuzzbench by updates to that commit hash.

I think having some comments on what changed in between experiments is important. There are two ways I want to do this.
1. We currently include the commit in each report we generate ([example](https://www.fuzzbench.com/reports/2020-04-14/index.html)). We could easily link to the diff on github between the current report's commit and the previous report's commit.
2. I'm thinking of introducing some kind of changelog/yaml file for fuzzbench that contributors update with the name of a fuzzer and a comment when they want a new experiment. Then some kind of automated system will start an experiment for the "changed" fuzzers and generate a report with previous data for the unchanged fuzzers.
